geonode/harvesting/utils.py

```diff
@@ -53,3 +52,25 @@ def validate_worker_configuration(harvester_type, configuration: typing.Dict):
             jsonschema.validate(configuration, schema)
         except jsonschema.exceptions.SchemaError as exc:
             raise RuntimeError(f"Invalid schema: {exc}")
+
+
+def get_xpath_value(
+        element: etree.Element,
+        xpath_expression: str,
+        nsmap: typing.Optional[dict] = None
+) -> typing.Optional[str]:
+    if not nsmap:
+        nsmap = element.nsmap
+    values = element.xpath(f"{xpath_expression}//text()", namespaces=nsmap)
+    return "".join(values).strip() or None
+
+
+def get_xpath_elements(
+        element: etree.Element,
+        xpath_expression: str,
+        nsmap: typing.Optional[dict] = None
+) -> typing.List[etree.Element]:
```

Comment:
If you can always get `nsmap` from the provided `element`, why have it also as an optional parameter? Is there an actual usecase for this in the code?